packages/editor/src/client/panel/imageEditPanel.tsx

```diff
@@ -193,6 +195,62 @@ export function ImagedEditPanel({id, file, onClose, onSave}: ImageEditPanelProps
     }
   }
 
+  /**
+   * Resizes an image on client side, if larger than 10 MB, which is the upload max of the api
+   * @param file
+   */
+  async function resizeImage(file: File): Promise<File> {
+    // only resize image if larger than 10 MB
+    if (!willImageResize(file)) {
+      return file // do not resize
+    }
+    const originalFileName: string = file.name
+    if (!originalFileName) {
+      throw new Error('No file name provided!')
+    }
+    const options = {
+      targetSize: 10, // the max size in MB, defaults to 2MB
+      quality: 1, // the quality of the image, max is 1,
+      resize: false // defaults to true, set false if you do not want to resize the image width and height
+    }
+    const compressor = new Compress(options)
+    const conversions = await compressor.compress([file])
+    const {photo} = conversions[0]
+    const imgBlob: Blob = photo.data
+    // convert blob to file
+    return new File([imgBlob], originalFileName, {type: photo.type})
+  }
+
+  /**
+   * Decide whether an image will be automatically resized or not. The limit is 10 MB which
+   * is the api's max image size
+   * @param file
+   */
+  function willImageResize(file: File) {
+    const originalFileSize: number = file.size / (1024 * 1024)
+    const envMinSize: string | undefined = process.env.IMG_MIN_SIZE_TO_RESIZE
```

Comment:
This doesn't work. In the Client/Browser context you don't have access to `process`. You have to pass the variable in the server context `packages/editor/src/server/index.tsx` to the `clientSettings` on line 28. Afterwards you can access the variable like this:
```
const {variableName}: ClientSettings = JSON.parse(
    document.getElementById(ElementID.Settings)!.textContent!
  )
```

Reply:
yeah. true!